packages/core/src/command/hl7v2-subscriptions/hl7v2-to-fhir-conversion/adt/utils.ts

```diff
@@ -0,0 +1,114 @@
+import { buildPeriod } from "../../../../external/fhir/shared/timestamps";
+import { uuidv7 } from "../../../../util/uuid-v7";
+import {
+  getOptionalValueFromField,
+  getOptionalValueFromMessage,
+  getOptionalValueFromSegment,
+  getSegmentByNameOrFail,
+  mapHl7SystemNameToSystemUrl,
+} from "../shared";
+import { buildConditionCoding } from "./condition";
+
+type ConditionOffset = 0 | 3;
+
+type HumanNameDetails = {
+  family: string;
+  given: string;
+  secondaryGivenNames: string | undefined;
+  suffix: string | undefined;
+  prefix: string | undefined;
+};
+
+export function getPeriodFromPatientVisit(adt: Hl7Message): Encounter["period"] | undefined {
+  const pv1Segment = adt.getSegment("PV1");
+  if (!pv1Segment) return undefined;
+
+  const start = getOptionalValueFromSegment(pv1Segment, 44, 1);
+  const end = getOptionalValueFromSegment(pv1Segment, 45, 1);
+
+  return buildPeriod(start, end);
+}
+
+export function getPatientClassCode(adt: Hl7Message): string | undefined {
+  return getOptionalValueFromMessage(adt, "PV1", 2, 1);
+}
+
+export function getFacilityName(adt: Hl7Message): string | undefined {
+  const evn = getSegmentByNameOrFail(adt, "EVN");
+  const eventFacilityNamespace = getOptionalValueFromSegment(evn, 7, 1);
+  const eventFacilityUniversalId = getOptionalValueFromSegment(evn, 7, 2);
+  if (eventFacilityNamespace || eventFacilityUniversalId) {
+    return [eventFacilityNamespace, eventFacilityUniversalId].filter(Boolean).join(" - ");
+  }
+
+  const pv1Segment = adt.getSegment("PV1");
+  if (!pv1Segment) return undefined;
+
+  const servicingFacilityName = getOptionalValueFromSegment(pv1Segment, 39, 1);
+  if (servicingFacilityName) return servicingFacilityName;
+
+  const assignedPatientLocationFacility = getOptionalValueFromSegment(pv1Segment, 3, 4);
+  return assignedPatientLocationFacility ?? undefined;
+}
+
+export function getAttendingDoctorNameDetails(adt: Hl7Message): HumanNameDetails | undefined {
+  const pv1Segment = adt.getSegment("PV1");
+  if (!pv1Segment) return undefined;
+
+  const attendingDoctorField = pv1Segment.getField(7);
+  if (attendingDoctorField.components.length < 1) return undefined;
+
+  const family = attendingDoctorField.getComponent(2);
+  const given = attendingDoctorField.getComponent(3);
+  const secondaryGivenNames = getOptionalValueFromField(attendingDoctorField, 4);
+  const suffix = getOptionalValueFromField(attendingDoctorField, 5);
+  const prefix = getOptionalValueFromField(attendingDoctorField, 6);
+
+  return { family, given, secondaryGivenNames, suffix, prefix };
+}
+
+export function getConditionCoding(
+  pv2Segment: Hl7Segment,
+  offset: ConditionOffset = 0
+): Coding | undefined {
+  const conditionCode = getOptionalValueFromSegment(pv2Segment, 3, 1 + offset);
+  const conditionDisplay = getOptionalValueFromSegment(pv2Segment, 3, 2 + offset);
+  const conditionSystem = getOptionalValueFromSegment(pv2Segment, 3, 3 + offset);
```

Comment:
can we multiply the offset by 3 so that the function used from the outside is a bit more obvious?
getConditionCoding(pv2, 0)
getConditionCoding(pv2, 1)